### ISSUE TITLE
1817 remove companies that have no further purpose

### ISSUE THREAD

**Issue by ChrisTheCat** [tobymao/18xx#2459]
These companies have no purpose once used and could be removed from the game to reduce visual clutter in an already visually complex game:

Coal mines after all their tokens have been place
Train station after the station token is received on charter
Pittsburgh Steel Mill after a yellow tile is placed in Pittsburgh
Bridges after all rivers have yellow tiles (unlikely to happen in practice)

**Comment by perwestling** [comment]
It should be noted that there exists companies (in some 18xx) that have multiple effects. Example in 18SJ the Gellivari Company has a two-tile-teleport-lay that it can use once, and then a discount for specific tile lays. 
So it might be hard to make a general solution.

But having them be removed in 1817 might be a good enhancement.

**Comment by ChrisTheCat** [comment]
Yes, that is also true in 1817, where the bridge tokens can be consumed but
the river discount ability persists.

**Comment by jenf** [comment]
The train station has to persist until the end of the stock round where the corp acquires it... otherwise the code thinks that you no longer need to buy tokens.
